person.rb

```diff
@@ -2,31 +2,37 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = rand(1..100)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
   end
 
   def correct_name
     @name
   end
 
   def can_use_services?
-    of_age
+    of_age? || @parent_permission
   end
 
   def parent_permission?
     @parent_permission
   end
 
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
+
```

Comment:
- Hey :wave: in our method, we need two parameters, person, and date, to keep track of the date and the book this person has rented, which establishes the has-many relationship. Also, when adding a rental, the person itself should also be sent to the parameter to establish the has-many relationship. so our method now should be this below, kindly update yours



```ruby
def add_rental(book, date)
    Rental.new(date, self, book)
  end
```